tests/unit/test_ignore_stop_commands.py

```diff
@@ -0,0 +1,95 @@
+from ast import Tuple
+from enum import Enum
+from typing import Callable
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pyparsing import Any
+
+from autogpt.agents import Agent
+from autogpt.memory.message_history import MessageHistory
+from autogpt.spinner import Spinner
+from tests.unit.test_utils import testconfig, valid_json_response  # NOQA
+
+
+async def async_task_and_spin(spn: Any, some_task: Callable, args: Tuple) -> Any:
+    emulated_response_from_openai = (
+        "```" + str(async_task_and_spin.myresp) + "```"  # type: ignore
+    )
+    reply = MagicMock()
+    reply.content = emulated_response_from_openai
+    return reply
+
+
+class TestMode(Enum):
+    StopCommand = 0
+    IgnoreCommand = 1
+
+
+@pytest.mark.parametrize("test_type", [TestMode.StopCommand, TestMode.IgnoreCommand])
+def test_commands_to_ignore(
+    testconfig: Any, valid_json_response: Any, test_type: Any, mocker: Any
+) -> None:
+    def set_continous(*args: Any, **kwargs: Any) -> None:
+        # We want to stop after one command
+        config.continuous_mode = True
+        config.continuous_limit = 1
+
+    async_task_and_spin.myresp = valid_json_response  # type: ignore
+    # Assuming that 'ignore_this_command' should be ignored
+    config = testconfig
+
+    if test_type == TestMode.IgnoreCommand:
+        config.commands_to_ignore = ["ignore_this_command"]
+
+        # Mocking the methods to simulate the interaction loop
+        mocker.patch(
+            "autogpt.agents.agent.extract_command",
+            return_value=("ignore_this_command", {}),
+        )
+        next_action_count = 0
+    else:
+        config.commands_to_stop = ["stop_this_command"]
+        config.commands_to_ignore = []
+        next_action_count = 1
+        mocker.patch(
+            "autogpt.agents.agent.extract_command",
+            return_value=("stop_this_command", {}),
+        )
+
+    clean_input_mock = mocker.patch(
+        "autogpt.agents.agent.clean_input", return_value=config.authorise_key
+    )
+    ai_config = MagicMock()
+    ai_config.ai_name = "Test"
+
+    # with mock.patch("autogpt.workspace.workspace.Workspace", autospec=Workspace):
+    mocker.patch("autogpt.agents.agent.execute_command", side_effect=set_continous)
+    # For some reason config stopped working if it was a decorator
+    with patch.object(
```

Comment:
As mentioned, for some bloody reason , didn't work  as decorators on the function.